modules/contact/include/constraints/ComputeFrictionalForceLMMechanicalContact.h

```diff
@@ -49,26 +49,40 @@ class ComputeFrictionalForceLMMechanicalContact : public ComputeWeightedGapLMMec
    */
   virtual void enforceConstraintOnDof(const DofObject * const dof) override;
 
+  /**
+   * Method called from \p post(). Used to enforce node-associated constraints. E.g. for the base \p
+   * ComputeFrictionalForceLMMechanicalContact we enforce PDASS frictional constraints. This is also
+   * where we actually feed the node-based constraint information into the system residual and
+   * Jacobian. Method used to enforce frictional constraints in three-dimensional simulations.
+   */
+  virtual void enforceConstraintOnDof3d(const DofObject * const dof);
```

Comment:
I was hoping that this method would have disappeared

Reply:
The equations for 2d and 3d are quite different, I'd say completely different. I could still merge those two methods if you are so inclined.

Reply:
Ok then let's not merge them. But per my comment in the .C file, sans the ternary to determine the number of tangents (since we're keeping this method, we know it will be 2), let's make things a loop if we can such that we can get rid of all the `_dir`s